# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/PerformanceObserver.h
# Commit: 5248abfed275
# Full Hash: 5248abfed275b2938ed64770bec58b3ff68d624b
# Author: Will Hawkins <whawkins@mozilla.com>
# Date: 2019-04-12 21:56:33
# Regressor Bug: 1539006
# File Overlap Count: 1
# Description:
#   Bug 1539006: Properly implement Performance Timeline Level 2 w3c spec. r=mstange,baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27008
# ==============================================================================

diff -r e17500911824 -r 5248abfed275 dom/performance/PerformanceObserver.h
--- a/dom/performance/PerformanceObserver.h	Fri Apr 12 15:49:22 2019 +0000
+++ b/dom/performance/PerformanceObserver.h	Fri Apr 12 16:29:48 2019 +0000
@@ -49,7 +49,9 @@
 
   nsISupports* GetParentObject() const { return mOwner; }
 
-  void Observe(const PerformanceObserverInit& aOptions);
+  void Observe(const PerformanceObserverInit& aOptions, ErrorResult& aRv);
+  static void GetSupportedEntryTypes(const GlobalObject& aGlobal,
+                                     JS::MutableHandle<JSObject*> aObject);
 
   void Disconnect();
 
@@ -58,13 +60,27 @@
   MOZ_CAN_RUN_SCRIPT void Notify();
   void QueueEntry(PerformanceEntry* aEntry);
 
+  bool ObservesTypeOfEntry(PerformanceEntry* aEntry);
+
  private:
+  void ReportUnsupportedTypesErrorToConsole(bool aIsMainThread,
+                                            const char* msgId,
+                                            const nsString& aInvalidTypes);
   ~PerformanceObserver();
 
   nsCOMPtr<nsISupports> mOwner;
   RefPtr<PerformanceObserverCallback> mCallback;
   RefPtr<Performance> mPerformance;
   nsTArray<nsString> mEntryTypes;
+  nsTArray<PerformanceObserverInit> mOptions;
+  enum {
+    ObserverTypeUndefined,
+    ObserverTypeSingle,
+    ObserverTypeMultiple,
+  } mObserverType;
+  /*
+   * This is also known as registered, in the spec.
+   */
   bool mConnected;
   nsTArray<RefPtr<PerformanceEntry>> mQueuedEntries;
 };